syntax = "proto3";
import "google/protobuf/empty.proto";
option go_package = ".;proto";

service User {
  rpc GetUserList(PageInfoRequest) returns (UserListResponse); // 用户列表
  rpc GetUserByMobile(MobileRequest) returns (UserInfoResponse); // 通过 mobile 查询用户
  rpc GetUserById(IdRequest) returns (UserInfoResponse); // 通过 ID 查询用户
  rpc CreateUser(CreateUserRequest) returns (UserInfoResponse); // 添加用户
  rpc UpdateUser(UpdateUserRequest) returns (google.protobuf.Empty); // 更新用户
}

message PageInfoRequest {
  uint32 page = 1;
  uint32 size = 2;
}

message MobileRequest {
  string mobile = 1;
}

message IdRequest {
  int64 id = 1;
}

message CreateUserRequest {
  string nickName = 1;
  string password = 2;
  string mobile = 3;
}

message UpdateUserRequest {
  int64 id = 1;
  string nickName = 2;
  uint32 gender = 3;
  uint64 birthday = 4;
}

message UserInfoResponse {
  int64 id = 1;
  string mobile = 2;
  string password = 3;
  string nickName = 4;
  int32 gender = 5;
  uint64 birthday = 6;
  int32 role = 7;

}

message UserListResponse {
  repeated UserInfoResponse data = 2;
}
